rust/src/encodings/plain.rs

```diff
@@ -725,4 +750,33 @@ mod tests {
             &BooleanArray::from(vec![false, false, true, false, true])
         );
     }
+
+    #[tokio::test]
+    async fn test_take_boolean_beyond_chunk() {
+        let store = ObjectStore::memory_small_prefetch();
```

Comment:
where does this function come from

Reply:
I found it have the same  effect with 
```
        let store = ObjectStore::memory();
        store.set_prefetch_size(256);
```
Use that one instead.

The full test function is for test if the take_boolean works well if the range is larger than a chunk.